### PR TITLE
Flp 77 create query store

### PR DESCRIPTION
**Pull request by MichaelYoung1981**

### CODE REVIEW

File: src/Infrastructure/SqlAzureIdentityAuthenticationDbConnectionInterceptor.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        if (_connectionNeedsAccessToken)
+        {
+            _logger.LogInformation("Getting AccessToken");
+            var token = _tokenProvider.GetAccessToken();
+            sqlConnection.AccessToken = token;
+        }
+        else
+        {
+            _logger.LogWarning($"Skipping GetAccessToken because ConnectionNeedsAccessToken is {_connectionNeedsAccessToken}");
+        }
```
Comment (ruslanss) [inline src/Infrastructure/SqlAzureIdentityAuthenticationDbConnectionInterceptor.cs:36]: is this logging temporary?

File: src/Infrastructure/SqlAzureIdentityAuthenticationDbConnectionInterceptor.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            var token = await _tokenProvider.GetAccessTokenAsync();
+
+            sqlConnection.AccessToken = token;
+        }
+        else
+        {
+            _logger.LogWarning($"Skipping GetAccessToken Async because ConnectionNeedsAccessToken is {_connectionNeedsAccessToken}");
+        }
+
+        return await base.ConnectionOpeningAsync(connection, eventData, result, cancellationToken);
```
Comment (ruslanss) [inline src/Infrastructure/SqlAzureIdentityAuthenticationDbConnectionInterceptor.cs:61]: more logging. looks like someone was debugging deployment

File: src/Infrastructure/SqlAzureIdentityTokenProvider.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+        return token;
+    }
+
+    public string GetAccessToken()
+    {
+        var token = _azureServiceTokenProvider.GetAccessTokenAsync("https://database.windows.net/").GetAwaiter().GetResult();
+
+        _logger.LogInformation("Generated SQL AccessToken");
+
```
Comment (ruslanss) [inline src/Infrastructure/SqlAzureIdentityTokenProvider.cs:31]: more logging

File: src/InnerApi/SFA.DAS.Funding.ApprenticeshipEarnings.InnerApi.csproj [hunk 4]
```diff
@@ -18,4 +18,8 @@
     <ProjectReference Include="..\DataAccess\SFA.DAS.Funding.ApprenticeshipEarnings.DataAccess.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Controllers\" />
+  </ItemGroup>
+
```
Comment (ruslanss) [inline src/InnerApi/SFA.DAS.Funding.ApprenticeshipEarnings.InnerApi.csproj:24]: is this needed?

**Review by ruslanss — CHANGES_REQUESTED** [review]
check comments

**Review by ruslanss — APPROVED** [review]

**Review by tom-gough — APPROVED** [review]

**Pull request merged**